dashboard/serializers_test.py

```diff
@@ -442,48 +415,32 @@ def get_sorted_semester_values(serialized_program_user):
-        expected_semester_enrollment_values = [
-            '2016 - Fall',
-            '2016 - Spring',
-            '2015 - Fall',
-            '2015 - Summer',
-            '2015 - Spring'
-        ]
-        assert sorted_semester_enrollment_values == sorted(expected_semester_enrollment_values)
-
-    def test_multiple_semester_enrollments(self):
-        """
-        Tests that enrollments in multiple course runs with the same season/year will not yield
-        additional serialized semesters
-        """
-        serialized_program_user = UserProgramSearchSerializer.serialize(self.program_enrollment)
-        # Add a second Spring 2016 course run
-        course_run = CourseRunFactory.create(
-            course__program=self.program,
-            edx_course_key='course-v1:MITx+newcourse+1T2016'
-        )
-        CachedEnrollmentFactory.create(user=self.user, course_run=course_run)
-        new_serialized_program_user = UserProgramSearchSerializer.serialize(self.program_enrollment)
-        assert self.get_sorted_semester_values(serialized_program_user) == \
-            self.get_sorted_semester_values(new_serialized_program_user)
-
-    def test_enrollment_with_no_season_info(self):
-        """
-        Tests that an enrollment in a course run with insufficient year/season info won't be serialized
-        """
-        serialized_program_user = UserProgramSearchSerializer.serialize(self.program_enrollment)
-        # Add a course run that can't have year/season information parsed from it
-        course_run = CourseRunFactory.create(course__program=self.program, start_date=None, edx_course_key='bad_key')
-        CachedEnrollmentFactory.create(user=self.user, course_run=course_run)
-        new_serialized_program_user = UserProgramSearchSerializer.serialize(self.program_enrollment)
-        assert self.get_sorted_semester_values(serialized_program_user) == \
-            self.get_sorted_semester_values(new_serialized_program_user)
+            serialized['semester'] for serialized in serialized_program_user['enrollments']
+        ], key=lambda x: float('inf') if x is None else x)
+
+    def test_semester_serialization(self):
+        """
+        Tests that each course run has a string semester value as part of its serialization
+        """
+        num_courses = 5
+        courses = CourseFactory.create_batch(num_courses, program=self.program)
+        course_runs = [CourseRunFactory.create(course=course) for course in courses]
+        for course_run in course_runs:
+            CachedEnrollmentVerifiedFactory.create(user=self.user, course_run=course_run)
+        with patch(
+            'dashboard.serializers.get_year_season_from_course_run', autospec=True, return_value=(2017, 'Spring')
+        ) as get_year_season_patch:
+            serialized_program_user = UserProgramSearchSerializer.serialize(self.program_enrollment)
+        assert len(serialized_program_user['enrollments']) == num_courses
```

Comment:
Can you add an assertion for the semester value in each enrollment? As it is I can change the `semester` value in `serialized_enrollment` without breaking any tests, since we're only checking the length here and the next test is on `serialize_semester`